src/arith/rewrite_simplify.cc

```diff
@@ -353,6 +353,15 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const SubNode* op) {
         truncdiv(x + c1, c3) - truncdiv(x, c3), truncdiv(truncmod(x, c3) + c1, c3),
         CanProveGreaterEqual(x.Eval(), 0) && c1.Eval()->value >= 0 && c3.Eval()->value > 0);
 
+    TVM_TRY_REWRITE_IF(truncdiv(x * c1 + c2, c3) - truncdiv(x * c1 + c4, c3), truncdiv(c2 - c4, c3),
+                       c1.Eval()->value >= 0 && c2.Eval()->value > 0 && c3.Eval()->value > 0 &&
```

Comment:
Would be great if we can write down a proof for the rule. or a tabular validation. This rule is quite similar to the rule above(which translates into an additional truncmod), but this rule does not.

Reply:
ok, I add some proof above, please check